From 4dbeb5c63c95cfa154bebba0cb1fd72a6edc5886 Mon Sep 17 00:00:00 2001
From: henry shao <hshao@netlogicmicro.com>
Date: Tue, 31 Aug 2010 03:38:07 -0700
Subject: [PATCH 119/565] clean up for irq.c common/smp.c

Clean up for irq.c common/smp.c.

Based on Broadcom SDK 2.3.

Signed-off-by: henry shao <hshao@netlogicmicro.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/netlogic/xlp/irq.c | 27 +++------------------------
 1 file changed, 3 insertions(+), 24 deletions(-)

diff --git a/arch/mips/netlogic/xlp/irq.c b/arch/mips/netlogic/xlp/irq.c
index 4109086..6ab54bf 100644
--- a/arch/mips/netlogic/xlp/irq.c
+++ b/arch/mips/netlogic/xlp/irq.c
@@ -365,41 +365,20 @@ void __init init_nlm_common_irqs(void)
 	nlm_xlp_irq_mask |= (1ULL << IRQ_TIMER);
 }
 
-#ifdef CONFIG_KGDB
-extern irqreturn_t xlp_kgdb_ipi_handler(int irq, struct pt_regs *regs);
-#endif
-#ifdef CONFIG_OPROFILE
-extern void nlm_common_oprofile_int_handler(int irq, void *dev_id,
-					 struct pt_regs *regs);
-#endif
-
 void do_nlm_common_IRQ(unsigned int irq, struct pt_regs *regs)
 {
-#ifdef CONFIG_SMP
-
-#ifdef CONFIG_NLMCOMMON_IP_FLOW_AFFINITY
-	if (irq == IRQ_IPI_SMP_FUNCTION || irq == IRQ_IPI_SMP_RESCHEDULE
-	    || irq == IRQ_IPI_NETRX) {
-#else
 	if (irq == IRQ_IPI_SMP_FUNCTION || irq == IRQ_IPI_SMP_RESCHEDULE) {
-#endif				/* CONFIG_NLMCOMMON_IP_FLOW_AFFINITY */
 		nlm_common_ipi_handler(irq, regs);
 		return;
 	}
-#endif
 	if (irq == IRQ_MSGRING) nlm_xlp_msgring_int_handler(irq, regs);
 
-#ifdef CONFIG_KGDB
 	else if (irq == IRQ_IPI_SMP_KGDB) {
-  }
-#endif
-#ifdef CONFIG_OPROFILE
+  	}
 	else if (irq == IRQ_IPI_OPROFILE) {
-		if (netlogic_thr_id() != 0)
-			nlm_common_oprofile_int_handler(irq, NULL, regs);
 	}
-#endif
-  else do_IRQ(irq);
+  	else 
+		do_IRQ(irq);
 }
 
 void __cpuinit nlm_smp_irq_init(void)
-- 
1.8.4.93.g57e4c17

